c_src/erl_optics.c

```diff
@@ -511,13 +546,18 @@ static ERL_NIF_TERM eo_optics_poll(
     if (!optics) return ERROR("get_optics");
     if (!poller) return ERROR("poller_allocation_error");
 
-    ctx->env = env;
-    ctx->map = enif_make_new_map(env);
-
+    if (erlang_poller){
+      assert(ctx);
+      ctx->env = env;
+      ctx->map = enif_make_new_map(env);
+      optics_poller_poll(poller);
+      return enif_make_tuple2(env, atom_ok, ctx->map);
```

Comment:
Seems like this is mis-indented as there's 6 spaces and that's just a wrong number of spaces :)
See my previous comment about automating things like formatting.